### PR TITLE
Prompt [Y/n] when linking key if only one found in GPG

### PR DESCRIPTION
**Pull request by idrysdale**

### CODE REVIEW

**Comment by paulfurley** [comment]
Mergefail! :(

File: fluidkeys/main.go [hunk 1]
```diff
@@ -115,17 +115,21 @@ func keyFromGpg() exitCode {
...
-	fmt.Printf("Key to import: %v", keyToImport.Fingerprint)
+	if keyToImport != nil {
+		fmt.Printf("Key to import: %v", keyToImport.Fingerprint)
 
-	fluidkeysDirectory, err := getFluidkeysDirectory()
-	if err != nil {
-		fmt.Printf("Failed to get fluidkeys directory")
-		return 1
-	}
+		fluidkeysDirectory, err := getFluidkeysDirectory()
```
Comment (paulfurley) [inline fluidkeys/main.go]: tabs + spaces, make yer mind up guv!

File: fluidkeys/main.go [hunk 2]
```diff
@@ -115,15 +115,21 @@ func keyFromGpg() exitCode {
 	fmt.Printf(formatListedKeysForImportingFromGpg(secretKeys))
 	keyToImport := promptForKeyToImportFromGpg(secretKeys)
 
-	fluidkeysDirectory, err := getFluidkeysDirectory()
-	if err != nil {
-		fmt.Printf("Failed to get fluidkeys directory")
-		return 1
-	}
+	if keyToImport != nil {
```
[the remark this answers is not in the document]
Comment (idrysdale) [reply fluidkeys/main.go]: Good shout

**Comment by idrysdale** [comment]
Are you done reviewing this @paulfurley ?

I've made some changes in line with the review.

**Comment by paulfurley** [comment]
Not finished no

File: fluidkeys/main.go [hunk 3]
```diff
@@ -254,24 +259,57 @@ func printSecretKeyListing(listNumber int, key gpgwrapper.SecretKeyListing) stri
...
 			}
 		}
+		return &secretKeyListings[selectedKey]
+	}
+}
+
+func promptToConfirmImportKeyFromGpg(key gpgwrapper.SecretKeyListing) bool {
+	var confirmed bool
+	for validInput := false; !validInput; {
+		input := promptForInput("Import key? [Y/n] ")
```
Comment (paulfurley) [inline fluidkeys/main.go]: This advertises a default of `Y` (by virtue of it being uppercase) but it *appears* the switch doesn't support just hitting `[return]`, is that right?

File: fluidkeys/main.go [hunk 4]
```diff
@@ -254,24 +259,57 @@ func printSecretKeyListing(listNumber int, key gpgwrapper.SecretKeyListing) stri
...
+		switch input {
+		case "Y":
```
Comment (paulfurley) [inline fluidkeys/main.go]: Any reason why not `switch strings.Lower(input)`?

File: fluidkeys/main.go [hunk 5]
```diff
@@ -254,24 +259,57 @@ func printSecretKeyListing(listNumber int, key gpgwrapper.SecretKeyListing) stri
...
+			validInput = true
+		case "n":
+			confirmed = false
+			validInput = true
+		default:
+			fmt.Printf("Please select only Y or N.\n")
+		}
 	}
-	return secretKeyListings[selectedKey]
+	return confirmed
```
Comment (paulfurley) [inline fluidkeys/main.go]: You could lose 2 bits of state and massively simplify this by simply returning:
```
switch stringsToLower(input) {
case "y":
    return true
case "n":
    return false
default:
   ...
}
```

File: fluidkeys/main.go (hunk 3 above)
Comment (idrysdale) [reply fluidkeys/main.go]: Shit, good spot!

File: fluidkeys/main.go (hunk 4 above)
Comment (idrysdale) [reply fluidkeys/main.go]: 👍

**Comment by paulfurley** [comment]
All done!

File: fluidkeys/main.go (hunk 5 above)
Comment (idrysdale) [reply fluidkeys/main.go]: That's nice.

**Comment by idrysdale** [comment]
Cool thanks for that. Some good pointers in there.

I've made changes in line with your suggestions.

**Pull request merged**